pkg/apis/pipeline/v1beta1/task_validation_test.go

```diff
@@ -436,6 +436,54 @@ func TestTaskSpecValidate(t *testing.T) {
 	}
 }
 
+func TestTaskSpecResultsValidate(t *testing.T) {
```

Comment:
NIT: I guess this really could be a `result_validation_test.go`, there's nothing relevant to the tests in the steps part oof the `Task` tested.

Reply:
Yeah sure! I was actually hesitant to add a new file(result_validation_test.go) or not. This test is testing from an upper level--testing the taskspec's validation and it will call result validation.

Reply:
I can create a new file and put the test there. Maybe in another PR add more tests?

Reply:
It would be great to have them in one PR since I need to back-port this on v0.36 an v0.37, but I can also backport the two PRs if you prefer so

Reply:
sure, I will add it in this pr!